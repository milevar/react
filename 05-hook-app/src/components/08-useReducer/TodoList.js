import React from 'react';
import TodoListItem from './TodoListItem';

const TodoList = ({todos, handleDelete, handleToggle}) => {
    return (
        <ul className="list-group list-group-flush">
            {
                todos.map((todo, i) => (
                    <TodoListItem
                        key={todo.id}
                        todo={todo}
                        index={i}
                        handleDelete={() => handleDelete(todo.id)}
                        handleToggle={() => handleToggle(todo.id)}/>
                ))
            }
        </ul>
    );
};

export default TodoList;
