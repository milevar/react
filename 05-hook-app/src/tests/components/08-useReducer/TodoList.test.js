import React from "react";
import {shallow} from 'enzyme';


import {demoTodos} from "../../fixtures/demoTodos";
import TodoList from "../../../components/08-useReducer/TodoList";


describe('Pruebas en <TodoList />', () => {

    const handleDelete = jest.fn();
    const handleToggle = jest.fn();

    const wrapper = shallow(<
        TodoList
        todos={ demoTodos}
        handleDelete={handleDelete}
        handleToggle={handleToggle}
    />);

    test('debe mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('debe tener 2 elementos', () => {

        expect(wrapper.find('TodoListItem').length).toBe(demoTodos.length);
        expect(wrapper.find('TodoListItem').at(0).prop('handleDelete')).toEqual(expect.any(Function));
    });

});
