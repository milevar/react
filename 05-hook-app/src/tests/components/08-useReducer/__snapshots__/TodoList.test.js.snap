// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <TodoList /> debe mostrarse correctamente 1`] = `
<ul
  className="list-group list-group-flush"
>
  <TodoListItem
    handleDelete={[Function]}
    handleToggle={[Function]}
    index={0}
    key="1"
    todo={
      Object {
        "desc": " Aprender React",
        "done": false,
        "id": 1,
      }
    }
  />
  <TodoListItem
    handleDelete={[Function]}
    handleToggle={[Function]}
    index={1}
    key="2"
    todo={
      Object {
        "desc": " Aprender Mongo",
        "done": false,
        "id": 2,
      }
    }
  />
</ul>
`;
